packages/jetbrains-plugin/src/test/kotlin/com/mongodb/jbplugin/fixtures/IntegrationTestExtensions.kt

```diff
@@ -0,0 +1,153 @@
+ */
+private class IntegrationTestExtension : BeforeTestExecutionCallback,
+    AfterTestExecutionCallback,
+    ParameterResolver {
+    private lateinit var application: Application
+    private lateinit var messageBus: MessageBus
+    private lateinit var project: Project
+
+    override fun beforeTestExecution(context: ExtensionContext?) {
+        application = mock()
+        project = mock()
+
+        messageBus = RootBus(mock<MessageBusOwner>().apply {
+            `when`(this.isDisposed()).thenReturn(false)
+            `when`(this.isParentLazyListenersIgnored()).thenReturn(false)
+            `when`(this.createListener(any())).then {
+                val descriptor = it.arguments[0] as ListenerDescriptor
+                Class.forName(descriptor.listenerClassName).getConstructor().newInstance()
+            }
+        })
+
+        `when`(application.getMessageBus()).thenReturn(messageBus)
+        `when`(project.getMessageBus()).thenReturn(messageBus)
+
+        ApplicationManager.setApplication(application) {}
+    }
+
+    override fun afterTestExecution(context: ExtensionContext?) {
+    }
+
+    override fun supportsParameter(parameterContext: ParameterContext?, extensionContext: ExtensionContext?): Boolean =
+        parameterContext?.parameter?.type?.run {
+            equals(Application::class.java) || equals(Project::class.java)
+        } ?: false
+
+    override fun resolveParameter(parameterContext: ParameterContext?, extensionContext: ExtensionContext?): Any =
+        when (parameterContext?.parameter?.type) {
+            Application::class.java -> application
+            Project::class.java -> project
+            else -> TODO()
+        }
+}
+
+/**
+ * Convenience function in application or project for tests. It mocks the implementation of a single service
+ * with whatever implementation is passed as a parameter. For example:
+ *
+ * ```kt
+ * application.withMockedService(mockRuntimeInformationService())
+ * project.withMockedService(mockRuntimeInformationService())
+ * ```
+ *
+ * @param serviceImpl
+ * @return itself so it can be chained
+ */
+inline fun <reified S : ComponentManager, reified T> S.withMockedService(serviceImpl: T): S {
+    `when`(getService(T::class.java)).thenReturn(serviceImpl)
+    return this
+}
+
+/**
+ * Generates a mock runtime information service, useful for testing. If you need
+ * to create your own. You'll likely will build first an information service and
+ * then inject it into a mock project, something like this:
+ *
+ * ```kt
+ * val myInfoService = mockRuntimeInformationService(userId = "hey")
+ * project.withMockedService(myInfoService)
+ * ```
+ *
+ * @param userId
+ * @param osName
+ * @param arch
+ * @param jvmVendor
+ * @param jvmVersion
+ * @param buildVersion
+ * @param applicationName
+ * @return A new mocked RuntimeInformationService
+ */
+internal fun mockRuntimeInformationService(
+    userId: String = "123456",
+    osName: String = "Winux OSX",
```

Comment:
😬

Reply:
I knew you were going to hate it and I'm proud of it.